src/RestControllers/AuthorizationController.php

```diff
@@ -140,97 +142,17 @@ private function configKeyPairs(): void
-                }
-                // third, store the keys on drive and store the passphrase in the database
-                file_put_contents($this->privateKey, $privkey);
-                chmod($this->privateKey, 0640);
-                file_put_contents($this->publicKey, $pubkey);
-                chmod($this->publicKey, 0660);
-                sqlStatementNoLog("INSERT INTO `keys` (`name`, `value`) VALUES ('oauth2passphrase', ?)", [$this->cryptoGen->encryptStandard($this->passphrase)]);
-            }
-            // confirm existence of passphrase
-            $pkey = sqlQueryNoLog("SELECT `name`, `value` FROM `keys` WHERE `name` = 'oauth2passphrase'");
-            if (!empty($pkey['name']) && ($pkey['name'] == 'oauth2passphrase')) {
-                $this->passphrase = $this->cryptoGen->decryptStandard($pkey['value']);
-                if (empty($this->passphrase)) {
-                    // if decrypted pssphrase is empty, then critical error and must exit
-                    $this->logger->error("OpenEMR error - oauth2 passphrase was blank after it was decrypted, so forced exit");
-                    throw OAuthServerException::serverError("Security error - problem with authorization server keys.");
-                }
-            } else {
-                // oauth2passphrase is missing so must exit
-                $this->logger->error("OpenEMR error - oauth2 passphrase is missing, so forced exit");
-                throw OAuthServerException::serverError("Security error - problem with authorization server keys.");
-            }
-            // confirm existence of key pair
-            if (!file_exists($this->privateKey) || !file_exists($this->publicKey)) {
-                // key pair is missing so must exit
-                $this->logger->error("OpenEMR error - oauth2 keypair is missing, so forced exit");
-                throw OAuthServerException::serverError("Security error - problem with authorization server keys.");
-            }
-        } catch (OAuthServerException $exception) {
+            $oauth2KeyConfig = new OAuth2KeyConfig($GLOBALS['OE_SITE_DIR']);
+            $oauth2KeyConfig->configKeyPairs();
+            $this->privateKey = $oauth2KeyConfig->getPrivateKeyLocation();
+            $this->publicKey = $oauth2KeyConfig->getPublicKeyLocation();
+            $this->oaEncryptionKey = $oauth2KeyConfig->getEncryptionKey();
+            $this->passphrase = $oauth2KeyConfig->getPassPhrase();
+        } catch (OAuth2KeyException $exception) {
+            $this->logger->error("OpenEMR error - " . $exception->getMessage() . ", so forced exit");
+            $serverException = OAuthServerException::serverError("Security error - problem with authorization server keys.", $exception);
             SessionUtil::oauthSessionCookieDestroy();
-            $this->emitResponse($exception->generateHttpResponse($response));
-            exit;
```

Comment:
very cool to see this stuff get refactored into a separate class. Just making sure it will still force an exit from script after catching the exception and emitting the response (noted the `exit;` was removed)